File: OcrFrontend/src/types/ocr.ts
export interface OCRRequest {
    file: File;
    lang: string;
    autocorrect: boolean;
    contrast: number;
    brightness: number;
    sharpness: number;
}

export interface NLPToken {
    text: string;
    pos: string;
    lemma: string;
}

export interface NLPAnalysis {
    entities: unknown[];
    tokens: NLPToken[];
    num_sentences: number;
}

export interface OCRResponse {
    text: string;
    corrected_text: string;
    nlp_analysis: NLPAnalysis;
    ai_corrected_text: string;
    summary: string;
    ai_score: number;
    error?: string;
    confidence?: number;
}

export interface PDFPageResult {
    page_number: number;
    text: string;
    summary: string | null;
    ai_score: number | null;
    ai_corrected_text: string | null;
}

export interface PDFAnalysisResponse {
    total_pages: number;
    pages: PDFPageResult[];
    overall_summary: string | null;
    overall_score: number | null;
    potential_enhancements: string | null;
}

export interface OCRSettings {
    lang: string;
    autocorrect: boolean;
    contrast: number;
    brightness: number;
    sharpness: number;
}

export const DEFAULT_OCR_SETTINGS: OCRSettings = {
    lang: 'eng',
    autocorrect: true,
    contrast: 1.8,
    brightness: 1.8,
    sharpness: 1.8,
};

File: app/schemas/ocr.py
from pydantic import BaseModel
from typing import List, Optional

class OCRRequest(BaseModel):
    lang: str = "eng"

class OCRResponse(BaseModel):
    text: str
    corrected_text: Optional[str] = None

class Entity(BaseModel):
    text: str
    label: str

class TokenInfo(BaseModel):
    text: str
    pos: str
    lemma: str

class NLPAnalysisResult(BaseModel):
    entities: List[Entity]
    tokens: List[TokenInfo]
    num_sentences: int

class FullAnalysisResponse(BaseModel):
    text: str
    corrected_text: Optional[str] = None
    nlp_analysis: Optional[NLPAnalysisResult] = None

class AIAnalysisResponse(BaseModel):
    text: str
    corrected_text: Optional[str] = None
    nlp_analysis: Optional[NLPAnalysisResult] = None
    ai_corrected_text: Optional[str] = None
    summary: Optional[str] = None
    ai_score: Optional[float] = None


class PageAnalysis(BaseModel):
    page_number: int
    text: str
    summary: Optional[str] = None
    ai_score: Optional[float] = None
    ai_corrected_text: Optional[str] = None

class PDFPageAnalysisResponse(BaseModel):
    total_pages: int
    pages: List[PageAnalysis]
    overall_summary: Optional[str] = None
    overall_score: Optional[float] = None
